sanic/response/types.py

```diff
@@ -106,9 +105,6 @@ def processed_headers(self) -> Iterator[Tuple[bytes, bytes]]:
         Returns:
             Iterator[Tuple[bytes, bytes]]: A list of header tuples encoded in bytes for sending
         """  # noqa: E501
-        # TODO: Make a blacklist set of header names and then filter with that
-        if self.status in (304, 412):  # Not Modified, Precondition Failed
-            self.headers = remove_entity_headers(self.headers)
```

Comment:
I feel like rather than removing this we need to more closely follow the standard and limit when we are removing the headers.

Reply:
> If the conditional GET used a strong cache validator (see [section](https://datatracker.ietf.org/doc/html/rfc2616#section-13.3.3)
   [13.3.3](https://datatracker.ietf.org/doc/html/rfc2616#section-13.3.3)), the response SHOULD NOT include other entity-headers.
   Otherwise (i.e., the conditional GET used a weak validator), the
   response MUST NOT include other entity-headers; this prevents
   inconsistencies between cached entity-bodies and updated headers.

Reply:
I have no interest to work further on this. Feel free to fork if you do.

Reply:
Failing test removed, passes OK now.